stdlib/public/core/Optional.swift

```diff
@@ -299,13 +299,19 @@ public // COMPILER_INTRINSIC
 func _diagnoseUnexpectedNilOptional(_filenameStart: Builtin.RawPointer,
                                     _filenameLength: Builtin.Word,
                                     _filenameIsASCII: Builtin.Int1,
-                                    _line: Builtin.Word) {
-  _preconditionFailure(
-    "unexpectedly found nil while unwrapping an Optional value",
-    file: StaticString(_start: _filenameStart,
-                       utf8CodeUnitCount: _filenameLength,
-                       isASCII: _filenameIsASCII),
-    line: UInt(_line))
+                                    _line: Builtin.Word,
+                                    _typeStart: Builtin.RawPointer,
+                                    _typeLength: Builtin.Word,
+                                    _typeIsASCII: Builtin.Int1) {
```

Comment:
The runtime knows how to stringify type pointers. You could pass the type metadata pointer here and hand it off to `swift_getTypeName` instead of generating it at compile time.